In [99]:
import pandas as pd
import numpy as np
from sklearn.utils import class_weight
from sklearn.utils import resample

df_raw = pd.read_csv("Payments.crdownload")

Let's take a look once more into our data set.

In [100]:
df_raw.describe()

step        amount  oldbalanceOrg  newbalanceOrig  \
count  5.500631e+06  5.500631e+06   5.500630e+06    5.500630e+06   
mean   2.075349e+02  1.812596e+05   8.411647e+05    8.630762e+05   
std    1.112426e+02  6.290423e+05   2.918859e+06    2.955706e+06   
min    1.000000e+00  0.000000e+00   0.000000e+00    0.000000e+00   
25%    1.380000e+02  1.335506e+04   0.000000e+00    0.000000e+00   
50%    2.120000e+02  7.550280e+04   1.386900e+04    0.000000e+00   
75%    3.020000e+02  2.095587e+05   1.073630e+05    1.449540e+05   
max    3.800000e+02  9.244552e+07   4.381886e+07    4.368662e+07   

       oldbalanceDest  newbalanceDest       isFraud  isFlaggedFraud  
count    5.500630e+06    5.500630e+06  5.500630e+06    5.500630e+06  
mean     1.084465e+06    1.213316e+06  7.680938e-04    5.453921e-07  
std      3.269174e+06    3.584992e+06  2.770386e-02    7.385065e-04  
min      0.000000e+00    0.000000e+00  0.000000e+00    0.000000e+00  
25%      0.000000e+00    0.000000e+00  0.000000e+00    0.000000e+00  
50%      1.362658e+05    2.191819e+05  0.000000e+00    0.000000e+00  
75%      9.467986e+05    1.118979e+06  0.000000e+00    0.000000e+00  
max      3.555534e+08    3.560159e+08  1.000000e+00    1.000000e+00

In [101]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5500631 entries, 0 to 5500630
Data columns (total 11 columns):
 #   Column          Dtype  
---  ------          -----  
 0   step            int64  
 1   type            object 
 2   amount          float64
 3   nameOrig        object 
 4   oldbalanceOrg   float64
 5   newbalanceOrig  float64
 6   nameDest        object 
 7   oldbalanceDest  float64
 8   newbalanceDest  float64
 9   isFraud         float64
 10  isFlaggedFraud  float64
dtypes: float64(7), int64(1), object(3)
memory usage: 461.6+ MB


In [102]:
df_raw.isnull().sum()

step              0
type              0
amount            0
nameOrig          0
oldbalanceOrg     1
newbalanceOrig    1
nameDest          1
oldbalanceDest    1
newbalanceDest    1
isFraud           1
isFlaggedFraud    1
dtype: int64

In [103]:
df = df_raw.dropna()

We will now drop all columns that are irrelevant to our modeling according to our discoveries in our EDA

In [104]:
df = df.drop("isFlaggedFraud", axis=1).drop("nameDest", axis=1).drop("nameOrig", axis=1)

We now need to take measures towards the columns represented by strings. Without some sort of transformation of those columns we will not be able to model our data. Due to the importance of the Type columns and its strong relation to Frauds I will be choosing Frequency Encoding as the frequency of certain transactions such as Cash Out and Transfer can be informative for our fraud prediction.

In [105]:
frequency_map = df['type'].value_counts().to_dict()
df['type'] = df['type'].map(frequency_map)

Now considering the computing power needed for the modeling we will be using a sample to dive forward into the modeling stage. Considering the entire data frame has 5500630 rows, I would like to use an approximate number to half of the data. In this case we will be using 30000.

In [106]:
df = df.sample(n=30000, replace=False)

In [109]:
df.to_csv("Wrangled_Payments.csv", index=False)